main.go

```diff
@@ -20,16 +21,55 @@ func main() {
 		defer f.Close()
 	}
 
-	err := logic.Setup()
+	parseErr := parse()
+	if parseErr != nil {
+		printHelp(parseErr)
+		os.Exit(1)
+	} else if *helpFlag {
+		printHelp(nil)
+		os.Exit(0)
+	}
+
+	err := logic.Setup(*fileArg)
+
+	if err == nil && subcommand != nil {
+		handleSubcommands()
+	}
 
 	if err == nil {
+		var model tea.Model
+		if subcommand != nil && *subcommand == "log" {
+			model, _ = log.InitializeModel(func() (tea.Model, tea.Cmd) {
+				return model, tea.Quit
+			})
+		} else {
+			model = startMenu.InitializeModel()
+		}
+
 		_, err = tea.
-			NewProgram(startMenu.InitializeModel()).
+			NewProgram(model).
 			Run()
 	}
 
-	if err != nil {
-		fmt.Println("Undskyld! Something went wrong >w< here it is: %v", err)
-		os.Exit(1)
+	handleIfError(err)
+}
+
+func handleSubcommands() {
+	switch *subcommand {
+	case "log":
+		if len(positionalArguments) == 0 {
+			// Handled in Bubbletea initialization code
+			return
+		}
+		handleIfError(log.Invoke(positionalArguments))
+	}
+	os.Exit(0)
+}
+
+func handleIfError(err error) {
+	if err == nil {
+		return
```

Comment:
```suggestion
	exitIfError(err)
}

func handleSubcommands() {
	switch *subcommand {
	case "log":
		if len(positionalArguments) == 0 {
			// Handled in Bubbletea initialization code
			return
		}
		exitIfError(log.Invoke(positionalArguments))
	}
	os.Exit(0)
}

func exitIfError(err error) {
	if err == nil {
		return
```

`handleIfError` isn't a very descriptive name. I suggest renaming the function to `exitIfError`